common/schema-builder/src/main/java/io/smallrye/graphql/schema/helper/TypeNameHelper.java

```diff
@@ -119,6 +120,18 @@ private static String getNameForClassType(ClassInfo classInfo, Annotations annot
         return sb.toString();
     }
 
+    public static String getMethodName(MethodInfo methodInfo, Annotations annotations) {
+        StringBuilder sb = new StringBuilder();
+
+        if (annotations.containsKeyAndValidValue(Annotations.NAME)) {
+            sb.append(annotations.getAnnotationValue(Annotations.NAME).asString().trim());
```

Comment:
Currently we only consider `@Name` annotation when renaming argument, but we could later on add some more cases if needed.